BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/Authentication/CredentialEnvelope.cs

```diff
@@ -3,25 +3,28 @@
     using System.Threading;
     using System.Threading.Tasks;
 
-    internal interface ICredentialEnvelope
+    /// <summary>
+    /// This interface defines a class that can interact with Azure.Core.TokenCredential.
+    /// </summary>
+    public abstract class CredentialEnvelope
     {
         /// <summary>
         /// Gets the TokenCredential instance held by this class.
         /// </summary>
-        object Credential { get; }
+        public abstract object Credential { get; }
```

Comment:
what is the use of `Credential`? Only GetToken/async is required by consumers, right?

Reply:
This provides the option of getting the `TokenCredential` object itself. 
I think profiler team was going to use this, but I can doublecheck.

Reply:
You can keep this internal.
We will call GetToken/GetTokenAsync.